lasagne/layers/local.py

```diff
@@ -0,0 +1,196 @@
+import theano.tensor as T
+
+from .. import init
+from .. import nonlinearities
+
+from .conv import Conv2DLayer
+
+
+__all__ = [
+    "LocallyConnected2DLayer",
+]
+
+
+class LocallyConnected2DLayer(Conv2DLayer):
+    """
+    lasagne.layers.LocallyConnected2DLayer(incoming, num_filters, filter_size,
+    stride=(1, 1), pad='same', untie_biases=False,
+    W=lasagne.init.GlorotUniform(), b=lasagne.init.Constant(0.),
+    nonlinearity=lasagne.nonlinearities.rectify, flip_filters=True,
+    channelwise=False, **kwargs)
+
+    2D locally connected layer
+
+    Performs an operation similar to a 2D convolution but without the weight
+    sharing, then optionally adds a bias and applies an elementwise
+    nonlinearity.
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape. The
+        output of this layer should be a 4D tensor, with shape
+        ``(batch_size, num_input_channels, input_rows, input_columns)``.
+
+    num_filters : int
+        The number of learnable convolutional filters this layer has.
+
+    filter_size : int or iterable of int
+        An integer or a 2-element tuple specifying the size of the filters.
+
+    stride : int or iterable of int
+        An integer or a 2-element tuple specifying the stride of the
+        convolution operation.
```

Comment:
Add a note that "This implementation only supports unit stride, the argument is provided for compatibility to convolutional layers only."